File: C++/Квадратное уравнение.cpp
#include <iostream>
#include <cmath>
using namespace std;
int main(){
	setlocale(LC_ALL, "Russian");
	float a,b,c,d,x1,x2;
	cin>>a>>b>>c;
	if (a==0){
		if (b==0) {
			cout<<"X - любое число";
	}
		else {
			cout<<"Решений нет";
		}
	}
	else{
	d=b*b-4*a*c;
		if (d<0)
			cout<<"Вещественных корней нет";
		else {
			if (d==0) {
			x1=-b*(2*a);
			x2=x1;
			cout<<x1<<" "<<x2;
		}
			else {
			x1=(-b+sqrt(d))/(2*a);
			x2=(-b-sqrt(d))/(2*a);
			cout<<x1<<" "<<x2;
		}
}
}	
}

File: Python/Квадратное уравнение.py
a=float(input("Введите a: "))
b=float(input("Введите b: "))
c=float(input("Введите c: "))
if a==0:
    if b==0:
        print("X - любое число")
    else:
        print("Решений нет")
else:
    d=b**2-4*a*c
    if d<0:
        print("Вещественных корней нет")
    else:
        if d==0:
            x1=-b*(2*a)
            x2=x1
            print(x1, " ", x2)
        else:
            import math
            x1=(-b+math.sqrt(d))/(2*a)
            x2=(-b-math.sqrt(d))/(2*a)
            print (x1, " ", x2)
